src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs

```diff
@@ -65,16 +71,28 @@ protected override void BeginProcessing()
 
         public override void ExecuteCmdlet()
         {
-            var workspaceId = this.ParameterSet.Equals(PropertiesParameterSetName) ? this.Id : this.Workspace.Id;
-            var updatedProperties = this.ParameterSet.Equals(PropertiesParameterSetName) ? new Workspace { Name = this.Name, Description = this.Description } : this.Workspace;
-
-            // The API will throw 400 saying that it "Cannot apply PATCH to navigation property users" if we don't null this property out
-            updatedProperties.Users = null;
+            if (this.ParameterSet.Equals(CapacityParameterSetName))
+            {
 
-            using (var client = this.CreateClient())
+                using (var client = this.CreateClient())
+                {
+                    client.Workspaces.MigrateWorkspaceCapacity(this.Id, this.CapacityId);
```

Comment:
Ideally we support workspace object being passed in the pipeline too but it's not a requirement

Reply:
Somehow I didn't get your comments and I merged it already. Let me open another PR to address this.